src/gap.c

```diff
@@ -1772,7 +1772,11 @@ void InitializeGap (
 #ifdef HPCGAP
     InitMainThread();
 #endif
-    InitGAPState(ActiveGAPState());
+    ModulesInitModuleState();
+    // FIXME: calling ModulesInitModuleState before InitLibrary has a weird effect:
+    // it means that the first time the `InitModuleState` function of a module is
+    // called is *before* its `InitLibrary` function has been run; but all other
+    // invocations, in other threads, of course are run *after* `InitLibrary`.
```

Comment:
Note: The issue described in the above comment is the main blocker for this PR, at least from my point of view. To resolve it, we may have to audit all `InitModuleState` functions, resp. all `InitLibrary functions of related modules.

However, it may very well be that some modules A, B are such that B's `InitLibrary` requires that A's `InitLibrary` *and* `InitModuleState` have already run. This suggest another, relatively simple fix: simply call `InitModuleState` right after each `InitLibrary`, from inside `ModulesInitLibrary`. I haven't tested this yet, so it may also run into problems; my point is more that there are more avenues to investigate.